SofaKernel/modules/SofaCore/src/sofa/core/ComponentNameHelper.cpp

```diff
@@ -0,0 +1,41 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sstream>
+#include <sofa/core/ComponentNameHelper.h>
+#include <sofa/core/ObjectFactory.h>
+
+namespace sofa::core
+{
+
+std::string ComponentNameHelper::resolveName(const std::string& type, const std::string& name)
+{
+    if (name.empty())
+    {
+        const std::string radix = sofa::core::ObjectFactory::ShortName(type);
```

Comment:
I'm not sure ObjectFactory is needed there as NameDecoder provide the shortName service.

Reply:
Yes, but it's not the same service.

- `ShortName` in `ObjectFactory` searches in the object factory. If found, it returns the class short name, which is set by `NameDecoder::getShortName<T>`. If not found, it returns an empty string.
- `NameDecoder::shortName` simply transforms the first letter in lower case.

To apply your suggestion, `NameDecoder::getShortName<T>` would be the ideal candidate, but unfortunately, the type is not available. `ComponentNameHelper` requires a `std::string`.

Reply:
To conclude: `ComponentNameHelper::resolveName` must not consult the object factory. It is the responsibility of the caller to provide a valid class name.